pkg/apis/autoscaling/v1alpha1/pa_types_test.go

```diff
@@ -332,6 +334,195 @@ func TestIsReady(t *testing.T) {
+			autoscaling.MaxScaleAnnotationKey: "100",
+		}),
+		wantMin: 1,
+		wantMax: 100,
+	}, {
+		name:    "absent",
+		pa:      pa(map[string]string{}),
+		wantMin: 0,
+		wantMax: 0,
+	}, {
+		name: "only min",
+		pa: pa(map[string]string{
+			autoscaling.MinScaleAnnotationKey: "1",
+		}),
+		wantMin: 1,
+		wantMax: 0,
+	}, {
+		name: "only max",
+		pa: pa(map[string]string{
+			autoscaling.MaxScaleAnnotationKey: "1",
+		}),
+		wantMin: 0,
+		wantMax: 1,
+	}, {
+		name: "malformed",
+		pa: pa(map[string]string{
+			autoscaling.MinScaleAnnotationKey: "ham",
+			autoscaling.MaxScaleAnnotationKey: "sandwich",
+		}),
+		wantMin: 0,
+		wantMax: 0,
+	}, {
+		name: "too small",
+		pa: pa(map[string]string{
+			autoscaling.MinScaleAnnotationKey: "-1",
+			autoscaling.MaxScaleAnnotationKey: "-1",
+		}),
+		wantMin: 0,
+		wantMax: 0,
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			min, max := tc.pa.ScaleBounds()
+			if min != tc.wantMin {
+				t.Errorf("%q expected min: %v got: %v", tc.name, tc.wantMin, min)
+			}
+			if max != tc.wantMax {
+				t.Errorf("%q expected max: %v got: %v", tc.name, tc.wantMax, max)
+			}
+		})
+	}
+}
+
+func TestMarkResourceNotOwned(t *testing.T) {
+	pa := pa(map[string]string{})
+	pa.Status.MarkResourceNotOwned("doesn't", "matter")
+	active := pa.Status.GetCondition("Active")
+	if active.Status != corev1.ConditionFalse {
+		t.Errorf("TestMarkResourceNotOwned expected active.Status: False got: %v", active.Status)
+	}
+	if active.Reason != "NotOwned" {
+		t.Errorf("TestMarkResourceNotOwned expected active.Reason: NotOwned got: %v", active.Reason)
+	}
+}
+
+func TestMarkResourceFailedCreation(t *testing.T) {
+	pa := pa(map[string]string{})
+	pa.Status.MarkResourceFailedCreation("doesn't", "matter")
+	active := pa.Status.GetCondition("Active")
+	if active.Status != corev1.ConditionFalse {
+		t.Errorf("TestMarkResourceFailedCreation expected active.Status: False got: %v", active.Status)
+	}
+	if active.Reason != "FailedCreate" {
+		t.Errorf("TestMarkResourceFailedCreation expected active.Reason: FailedCreate got: %v", active.Reason)
+	}
+}
+
+func TestClass(t *testing.T) {
+	cases := []struct {
+		name string
+		pa   *PodAutoscaler
+		want string
+	}{{
+		name: "kpa class",
+		pa: pa(map[string]string{
+			autoscaling.ClassAnnotationKey: autoscaling.KPA,
+		}),
+		want: "kpa.autoscaling.knative.dev",
+	}, {
+		name: "hpa class",
+		pa: pa(map[string]string{
+			autoscaling.ClassAnnotationKey: autoscaling.HPA,
+		}),
+		want: "hpa.autoscaling.knative.dev",
+	}, {
+		name: "default class",
+		pa:   pa(map[string]string{}),
+		want: "kpa.autoscaling.knative.dev",
+	}, {
+		name: "custom class",
+		pa: pa(map[string]string{
+			autoscaling.ClassAnnotationKey: "yolo.sandwich.com",
+		}),
+		want: "yolo.sandwich.com",
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.pa.Class()
+			if got != tc.want {
+				t.Errorf("%q expected class: %q got: %q", tc.name, tc.want, got)
```

Comment:
preference for got before want
https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

Reply:
Fixed.